vcr/stubs/httpx_stubs.py

```diff
@@ -25,11 +25,20 @@ def _transform_headers(httpx_reponse):
 
 
 def _to_serialized_response(httpx_reponse):
+    content = httpx_reponse.content
+    try:
+        if content is not None and not isinstance(content, str):
+            content = content.decode("utf-8")
+    except (TypeError, UnicodeDecodeError, AttributeError):
+        # Sometimes the string actually is binary or StringIO object,
+        # so if you can't decode it, just give up.
+        pass
```

Comment:
hey @yhlam , I think we could completely ignore this decode, as I did in my PR #583 

As you can see on [httpx](https://github.com/encode/httpx/blob/2d571046e15d3f4c4cbe5a68173cdf12e99cb3ca/httpx/_types.py#L77), the content attribute can be a `bytes` type.

Reply:
Yes, `httpx` can take `bytes`. That's why the `content` is encoded back to `bytes` in `_from_serialized_response` when constructing `httpx.Response`. The decode is for serializing the VCR response, not for `httpx`. If you take out the decode, I believe you will get `!!binary` in the serialized yaml even for plain text. I guess reason of decoding the data into unicode is trying to have human readable yaml file. This unicode / bytes conversion happens in other libraries as well, but it is done in for the [body](https://github.com/kevin1024/vcrpy/blob/c79a06f639dd628536c9868044e78df1012985f9/vcr/serializers/compat.py#L54) field only. Therefore, I suggest the better fix should be having a consistent serialization schema for all libraries, which is discussed in #463.